wdl-format/tests/format/ENCODE-DCC_chip-seq-pipeline/source.formatted.wdl

```diff
@@ -0,0 +1,3586 @@
+            runtime_environment = runtime_environment,
+        }
+    }
+    if (aligner_ != "bwa" && use_bwa_mem_for_pe) {
+        call raise_exception as error_use_bwa_mem_for_non_bwa { input:
+            msg = "To use chip.use_bwa_mem_for_pe, choose bwa for chip.aligner.",
+            runtime_environment = runtime_environment,
+        }
+    }
+    if (aligner_ != "bowtie2" && use_bowtie2_local_mode) {
+        call raise_exception as error_use_bowtie2_local_mode_for_non_bowtie2 { input:
+            msg = "To use chip.use_bowtie2_local_mode, choose bowtie2 for chip.aligner.",
+            runtime_environment = runtime_environment,
+        }
+    }
+    if (aligner_ == "custom" && (!defined(custom_align_py) || !defined(custom_aligner_idx_tar))) {
+        call raise_exception as error_custom_aligner { input:
+            msg = "To use a custom aligner, define chip.custom_align_py and chip.custom_aligner_idx_tar.",
+            runtime_environment = runtime_environment,
+        }
+    }
+
+    if ((ctl_depth_limit > 0 || exp_ctl_depth_ratio_limit > 0) && num_ctl > 1 && length(ctl_paired_ends) > 1) {
+        call raise_exception as error_subsample_pooled_control_with_mixed_endedness { input:
+            msg = "Cannot use automatic control subsampling (\"chip.ctl_depth_limit\">0 and \"chip.exp_ctl_depth_limit\">0) for " + "multiple controls with mixed endedness (e.g. SE ctl-rep1 and PE ctl-rep2). " + "Automatic control subsampling is enabled by default. " + "Disable automatic control subsampling by explicitly defining the above two parameters as 0 in your input JSON file. " + "You can still use manual control subsamping (\"chip.ctl_subsample_reads\">0) since it is done " + "for individual control's TAG-ALIGN output according to each control's endedness. ",
+            runtime_environment = runtime_environment,
+        }
+    }
+    if (pipeline_type == "control" && num_ctl > 0) {
+        call raise_exception as error_ctl_input_defined_in_control_mode { input:
+            msg = "In control mode (chip.pipeline_type: control), do not define ctl_* input variables. Define fastqs_repX_RY instead.",
+            runtime_environment = runtime_environment,
+        }
+    }
+    if (pipeline_type == "control" && num_rep_fastq == 0) {
+        call raise_exception as error_ctl_fastq_input_required_for_control_mode { input:
+            msg = "Control mode (chip.pipeline_type: control) is for FASTQs only. Define FASTQs in fastqs_repX_RY. Pipeline will recognize them as control FASTQs.",
+            runtime_environment = runtime_environment,
+        }
+    }
+
+    # align each replicate
+    scatter (i in range(num_rep)) {
+        # to override endedness definition for individual replicate
+        #     paired_end will override paired_ends[i]
+        Boolean paired_end_ = if !defined(paired_end) && i < length(paired_ends) then paired_ends[i] else select_first([
+            paired_end,
+        ])
```

Comment:
I think if an `if..then..else` gets line split at any point, then it makes sense to convert the whole thing to the multi-line structure.

Reply:
That's a good idea. But I'm not entirely sure how to do that with the current architecture 🤔 

The older messier implementation was pretty easy to build logic like that, because you could construct a sub-string and check it for newline characters, then handle the wrapping logic appropriately. The new design prevents "peeking" ahead at the result. There are `PreTokens` which should be mostly agnostic of those things, and then `PostTokens` that handle the minutiae. But `PostTokens` don't have the bigger picture context.

I'll have to think on this some.